explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
 ![Android home screen badge](images/android-badge.png)
 <br>Android home screen badge
 
-The purpose of this API is:
+Unlike document contexts, badges applied in "handle" contexts can be shown and
+updated even when there are no tabs or windows open for the app. So there are
+some special considerations for this use case, such as how to set the badge in
+response to a [Push message](https://www.w3.org/TR/push-api/), and how to scope
+a badge so that it appears on the desired set of pages or apps. The most
+commonly requested use case for "handle" badging is badging an app icon on the
+OS shelf, but it generalizes to non-installed sites as well.
 
-* To subtly notify the user that there is new activity that might require their attention without requiring an OS-level [notification](https://notifications.spec.whatwg.org/).
-* To indicate a small amount of additional information, such as an unread count.
-* To allow certain user blessed pages (such as Bookmarks or [Installed Web Applications](https://www.w3.org/TR/appmanifest/#installable-web-applications)) to convey this information, regardless of whether they are currently open.
+## Usage examples
 
-Non-goals are:
+The simplest possible usage of the API is a single call to `Badge.set` from a
+foreground context (which might be used to show an unread count in an email
+app):
 
-* To provide an arbitrary image badge. The  web platform already provides this capability via favicons.
+```js
+Badge.set(getUnreadCount());
+```
 
-Possible areas for expansion:
+This will set the badge for all pages and apps in the current origin until it is
+changed. If `getUnreadCount()` (the argument to `Badge.set`) is 0, it will
```

Comment:
Added that to the "Model" section. I said this:

> The user agent is allowed to clear the badge whenever there are no foreground pages open on the origin (the intention of this is so that when the user agent quits, it does not need to serialize all the badge data and restore it on start-up; sites should re-apply the badge when they open).

Is that OK? It's a bit hard to exactly define when UAs are allowed to clear badges.

Reply:
:+1: